src/genetic_forensic_portal/app/client/gf_api_client.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import annotations
+
+MISSING_DATA_ERROR = "data is required"
+SAMPLE_UUID = "this-is-a-uuid"
+NO_METADATA_UUID = "this-is-a-differentuuid"
+
+
+def upload_sample_analysis(data: bytes, metadata: str | None = None) -> str:
```

Comment:
```suggestion
def upload_sample_analysis(data: bytes, metadata: Optional[str] = None) -> str:
```

You can use `Optional` if the choices is a type and `None`. You can import from `typing` library `from typing import Optional`

Reply:
Trying to do that leads to the precommit hook for `ruff` to revert back to what I've posted, citing this rule https://docs.astral.sh/ruff/rules/non-pep604-annotation/ 

What's the preferred way to handle this? Follow the linter blindly or make it ignore this rule?

Reply:
discussed offline, we're going to go with what ruff has corrected things to